packages/vite/src/assets.ts

```diff
@@ -0,0 +1,40 @@
+import { ResolverLoader } from '@embroider/core';
+import type { Plugin } from 'vite';
+import * as process from 'process';
+import { dirname, join } from 'path';
+import { copyFileSync, mkdirpSync } from 'fs-extra';
+import glob from 'fast-glob';
+
+export function assets(): Plugin {
+  const cwd = process.cwd();
+  const resolverLoader = new ResolverLoader(cwd);
+  const engines = resolverLoader.resolver.options.engines;
+  return {
+    name: 'assets',
+    enforce: 'pre',
+    outputOptions(options) {
+      options.dir = join(process.cwd(), 'dist');
+    },
+    async writeBundle(options) {
+      const pubDir = join(process.cwd(), 'public');
+      const publicAppFiles = glob.sync('**/*', {
+        cwd: pubDir,
+      });
+      for (const publicAppFile of publicAppFiles) {
```

Comment:
we do not copy the app assets, because vite does bundle its own asset files. We only need the app public files